CMakeLists.txt

```diff
@@ -151,12 +151,14 @@ set(POSITION_INDEPENDENT_CODE True)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
-if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=apple-m1+crc")
-else()
-  set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -mavx2 -mfma -mavx -mf16c -mbmi2 -mbmi")
-endif()
+
+execute_process(
+  COMMAND
+    bash -c
+    "( source ./scripts/setup-helper-functions.sh; echo -n $(get_cxx_flags))"
+  OUTPUT_VARIABLE SCRIPT_CXX_FLAGS)
+
```

Comment:
A status message here to show SCRIPT_CXX_FLAGS (what architecture-specific flags are being used) will be helpful to debug

Reply:
FYI, You can run in --trace-source or --trace-expand mode to get variable values when debugging etc. But I will add this as you suggest.

Reply:
We can avoid it if the output is verbose. I am not very particular if `--trace-source` does the job.